applications/HDF5Application/python_scripts/utils.py

```diff
@@ -1,58 +1,36 @@
-'''HDF5 process utilities.
+"""HDF5 custom IO process utilities.
 
 license: HDF5Application/license.txt
-'''
-import KratosMultiphysics
-
-
-def IsDistributed():
-    return KratosMultiphysics.DataCommunicator.GetDefault().IsDistributed()
-
+"""
 
-def InsertSingleSetting(dest, key, parameters):
-    if dest.Has(key):
-        dest[key] = parameters
-    else:
-        dest.AddValue(key, parameters)
 
+__all__ = [
+    "DefaultSetter",
+    "ParametersWrapper",
+    "IsDistributed",
+    "CreateOperationSettings",
+]
 
-def InsertSettings(src, dest):
-    for key, parameters in src.items():
-        InsertSingleSetting(dest, key, parameters)
-
-
-def InsertArrayOfSettings(src, dest):
-    for current_src, current_dest in zip(src, dest):
-        InsertSettings(current_src, current_dest)
 
+import KratosMultiphysics
+from KratosMultiphysics.HDF5Application.core.utils import DefaultSetter
+from KratosMultiphysics.HDF5Application.core.utils import ParametersWrapper
 
-def CheckForDeprecatedFilename(deprecated_settings, module_name, *new_io_settings):
-    if deprecated_settings.Has("file_name"):
-        depr_msg = '\nDEPRECATION-WARNING: "file_name" should be specified in "file_settings". This will be removed in the future!\n'
-        KratosMultiphysics.Logger.PrintWarning(module_name, depr_msg)
-        file_name = deprecated_settings["file_name"].GetString().replace(
-            ".h5", "")
-        for io_settings in new_io_settings:
-            new_file_name = io_settings["file_name"].GetString().replace(
-                "<identifier>", file_name)
-            io_settings["file_name"].SetString(new_file_name)
 
+def IsDistributed():
+    return KratosMultiphysics.DataCommunicator.GetDefault().IsDistributed()
```

Comment:
Now that it is available I would suggest to check the ModelPart that is being used

Reply:
On my todo list for the next PR.